crates/criticaltrust/src/signatures/payload.rs

```diff
@@ -43,10 +43,12 @@ impl<T: Signable> SignedPayload<T> {
     }
 
     /// Add a new signature to this signed payload, generated using the provided [`KeyPair`].
-    pub fn add_signature(&mut self, keypair: &dyn KeyPair) -> Result<(), Error> {
+    pub async fn add_signature<K: KeyPair>(&mut self, keypair: &K) -> Result<(), Error> {
```

Comment:
This one of the main deviations in public API changes. We now mono-morphize this function since `KeyPair` is not longer able to be a trait object.